### ISSUE TITLE
anyOf + ref doesn't compile

### ISSUE THREAD

**Issue by hazam** [fastify/fast-json-stringify#189]
## 🐛 Bug Report

anyOf + ref doesn't compile. Ajv compiles the schema fine though

## To Reproduce

```js
require('fast-json-stringify')({
  type: 'object',
  properties: {
    'cs': {
      'anyOf': [
        {
          '$ref': '#/definitions/Option'
        },
        {
          'type': 'boolean'
        }
      ]
    }
  }, 'definitions': {
    'Option': {
      'type': 'string'
    }
  }
});

yields 

/fastify-bug-report/node_modules/fast-json-stringify/index.js:945
        throw new Error(`${schema.type} unsupported`)
```

## Expected behavior

Compile the schema - please note that anyOf without refs works fine.

## Your Environment

- *node version*: 10
- *fast-json-stringify version*: >=1.15.6
- *os*: Mac
- I tried also older fast-json-stringy versions (13) and got same problem

**Comment by hazam** [comment]
Edited the desc - got the fix as well, will send a PR soon

**Comment by hazam** [comment]
https://github.com/fastify/fast-json-stringify/pull/190

**Comment by mcollina** [comment]
Fixed in #207
